src/sat/sat_clause.cpp

```diff
@@ -153,14 +153,14 @@ namespace sat {
         ptr &= ~0xFFFFFFFFull; // Keep only high part
         unsigned i = 0;
         for (i = 0; i < m_num_segments; ++i)
-            if (m_segments[i] == ptr)
-                return i;
+          if (m_segments[i] == ptr) 
+            return i;
         i = m_num_segments;
 #if defined(Z3DEBUG)
         SASSERT(i < c_max_segments);
         if (i + 1 == c_max_segments) {
             m_overflow_valid = true;
-            i += c_max_segments * m_cls_offset2ptr.size();
+            i += c_max_segments * m_cls_offset2ptr.size() + 1;
```

Comment:
Why `+ 1`? In my understanding, the point of #715 was to reserve clause offsets such that `cl_off % c_max_segment == c_max_segment - 1` for map-based offsets to avoid collisions with traditional offsets that have segment number in their lower bits. To this end, the original increment preserved the remainder.